chia/wallet/nft_wallet/nft_off_chain.py

```diff
@@ -0,0 +1,115 @@
+import logging
+import sys
+from pathlib import Path
+from typing import Any, Dict, Optional
+
+import aiohttp
+
+from chia.types.blockchain_format.program import Program
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.util.default_root import DEFAULT_ROOT_PATH
+from chia.util.hash import std_hash
+from chia.wallet.nft_wallet.nft_info import NFTCoinInfo
+from chia.wallet.nft_wallet.uncurry_nft import UncurriedNFT
+
+DEFAULT_PREFIX_HASH_LENGTH = 3
+DEFAULT_CACHE_PATH = "./nft_cache"
+CACHE_PATH_KEY = "nft_metadata_cache_path"
+PREFIX_HASH_LENGTH_KEY = "nft_metadata_cache_hash_length"
+
+
+log = logging.getLogger(__name__)
+
+
+async def fetch_off_chain_metadata(nft_coin_info: NFTCoinInfo, ignore_size_limit: bool) -> Optional[str]:
+    uncurried_nft: Optional[UncurriedNFT] = UncurriedNFT.uncurry(*nft_coin_info.full_puzzle.uncurry())
+    if uncurried_nft is None:
+        log.error(f"Cannot fetch off-chain metadata, {nft_coin_info.nft_id} is not a NFT.")
+        return None
+    for uri in uncurried_nft.meta_uris.as_python():  # pylint: disable=E1133
+        async with aiohttp.ClientSession() as session:
+            async with session.get(str(uri, "utf-8")) as response:
```

Comment:
timeout by default is 5 minutes, which has a big impact to how long it could take to load all metadata (especially over time when NFTs will have a lot of obsolete metadata URLs). 30 seconds should be enough.